### PR TITLE
Add concurrent downloads to downloader.

### PR DESCRIPTION
**Pull request by s-l-teichmann**
Add `--worker` flag to downloader.

Some sites seems to limit the number of requests per time interval per http client.
This PR adds a flag to download more than one advisory at once via multiple
http clients to speed up the process. By default only two clients are used.

Furthermore this PR also adds the feature to stop the downloader by `Ctrl-C`ing.

When the `ignoresigcheck` flag is set the results of the signature checks are ignored.

Bumped Go to 1.20 to use `errors.Join`.

### CODE REVIEW

File: docs/csaf_downloader.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Application Options:
       --version                Display version of the binary
   -v, --verbose                Verbose output
   -r, --rate=                  The average upper limit of https operations per second
+  -w, --worker=                Number of concurrent downloads (default: 1)
```
Comment (JanHoefelmeyer) [inline docs/csaf_downloader.md:15]: Maybe add that using not positive or too large numbers will try to use the maximum amount of workers avaible. E.g. 

    If worker is set to 0 or less, the maximum amount of available workers will be used.

Down there with the two other explanatory lines?

File: cmd/csaf_downloader/main.go [hunk 2]
```diff
@@ -10,21 +10,26 @@
...
 	"log"
 	"net/http"
 	"os"
+	"os/signal"
 
 	"github.com/csaf-poc/csaf_distribution/util"
 	"github.com/jessevdk/go-flags"
 )
 
+const defaultWorker = 1
```
Comment (bernhardreiter) [inline cmd/csaf_downloader/main.go:24]: Shouldn't we use a default with at least 2 to demonstrate this advantage (and get more testing of it)?

File: cmd/csaf_downloader/downloader.go [hunk 3]
```diff
@@ -391,56 +462,54 @@ func (d *downloader) downloadFiles(label csaf.TLPLabel, files []csaf.AdvisoryFil
...
-	return nil
+func (d *downloader) mkdirAll(path string, perm os.FileMode) error {
+	d.mkdirMu.Lock()
+	defer d.mkdirMu.Unlock()
+	return os.MkdirAll(path, perm)
 }
 
-func (d *downloader) checkSignature(data []byte, sign *crypto.PGPSignature) bool {
+func (d *downloader) checkSignature(data []byte, sign *crypto.PGPSignature) error {
+	if d.opts.Insecure {
```
[the remark this answers is not in the document]
Comment (s-l-teichmann) [reply cmd/csaf_downloader/downloader.go]: I totally agree. Commit 3da3b4877889e20b7d691c8792fd70e0a186db01 introduces a special flag:
```
--ignoresigcheck         Ignore signature check results, just warn on mismatch
```

File: docs/csaf_downloader.md (hunk 1 above)
Comment (s-l-teichmann) [reply docs/csaf_downloader.md]: I gave that a second thought and I have removed the CPU stuff entirely.
In commit 3da3b4877889e20b7d691c8792fd70e0a186db01 the lower bound is now 1.

**Comment by s-l-teichmann** [comment]
> Shouldn't we use a default with at least 2 to demonstrate this advantage (and get more testing of it)?

We could ... I'm a bit undecided how conservative our default setting should be. @tschmidtb51 ?

**Review by JanHoefelmeyer — DISMISSED** [review]
LGTM

**Comment by tschmidtb51** [comment]
> Bumped Go to 1.20 to use `errors.Join`.

Please add an extra note for that to the release notes...

**Comment by tschmidtb51** [comment]
> We could ... I'm a bit undecided how conservative our default setting should be. @tschmidtb51 ?

In general, I liked the idea of doing it based on the available CPUs but I understand that this might be harder to explain... IMHO, we should set the default (at least) to `2` but allow to reduce it to `1`.

**Comment by s-l-teichmann** [comment]
> > Bumped Go to 1.20 to use `errors.Join`.
> 
> Please add an extra note for that to the release notes...

Will do.

**Comment by s-l-teichmann** [comment]
> > We could ... I'm a bit undecided how conservative our default setting should be. @tschmidtb51 ?
> 
> In general, I liked the idea of doing it based on the available CPUs but I understand that this might be harder to explain... IMHO, we should set the default (at least) to `2` but allow to reduce it to `1`.

Commit c98fafe0a1598c864519d9ff415d594aaa9b787c sets the default to two.

**Review by tschmidtb51 — APPROVED** [review]
LGTM and works

**Pull request merged**